core/src/test/java/jenkins/model/RunIdMigratorTest.java

```diff
@@ -83,13 +83,13 @@
 
     @Test public void legacy() throws Exception {
         assumeFalse("Symlinks don't work well on Windows", Functions.isWindows());
-        write("2014-01-02_03-04-05/build.xml", "<?xml version='1.0' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <number>99</number>\n  <otherstuff>ok</otherstuff>\n</run>");
+        write("2014-01-02_03-04-05/build.xml", "<?xml version='1.1' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <number>99</number>\n  <otherstuff>ok</otherstuff>\n</run>");
         link("99", "2014-01-02_03-04-05");
         link("lastFailedBuild", "-1");
         link("lastSuccessfulBuild", "99");
-        assertEquals("{2014-01-02_03-04-05={build.xml='<?xml version='1.0' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <number>99</number>\n  <otherstuff>ok</otherstuff>\n</run>'}, 99=→2014-01-02_03-04-05, lastFailedBuild=→-1, lastSuccessfulBuild=→99}", summarize());
+        assertEquals("{2014-01-02_03-04-05={build.xml='<?xml version='1.1' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <number>99</number>\n  <otherstuff>ok</otherstuff>\n</run>'}, 99=→2014-01-02_03-04-05, lastFailedBuild=→-1, lastSuccessfulBuild=→99}", summarize());
         assertTrue(migrator.migrate(dir, null));
-        assertEquals("{99={build.xml='<?xml version='1.0' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <id>2014-01-02_03-04-05</id>\n  <timestamp>1388649845000</timestamp>\n  <otherstuff>ok</otherstuff>\n</run>'}, lastFailedBuild=→-1, lastSuccessfulBuild=→99, legacyIds='2014-01-02_03-04-05 99\n'}", summarize());
+        assertEquals("{99={build.xml='<?xml version='1.1' encoding='UTF-8'?>\n<run>\n  <stuff>ok</stuff>\n  <id>2014-01-02_03-04-05</id>\n  <timestamp>1388649845000</timestamp>\n  <otherstuff>ok</otherstuff>\n</run>'}, lastFailedBuild=→-1, lastSuccessfulBuild=→99, legacyIds='2014-01-02_03-04-05 99\n'}", summarize());
         assertEquals(99, migrator.findNumber("2014-01-02_03-04-05"));
         migrator = new RunIdMigrator();
         assertFalse(migrator.migrate(dir, null));
```

Comment:
I agree with @jtnord IIUC think you are right to do that replacement, but only after that `migrator.migrate` call, which is expected to migrate things from old 1.0 to new.

There can't be a Jenkins version using xml 1.1, AND using old IDs.

Reply:
Stepping through the migrator code, its just doing basic File operations (no xstream) so in this case it doesn't matter about the XML version.  I will revert them back to '1.0'